worker/uniter/charm/bundles.go

```diff
@@ -131,8 +133,29 @@ func (d *BundlesDir) bundleURLPath(url *charm.URL) string {
 	return path.Join(d.path, charm.Quote(url.String()))
 }
 
+// ClearDownloads removes any entries in the temporary bundle download
+// directory. It is intended to be called on uniter startup.
+func ClearDownloads(bundlesDir string) (err error) {
+	defer errors.DeferredAnnotatef(&err, "unable to clear bundle downloads")
+
+	downloadDir := downloadsPath(bundlesDir)
```

Comment:
Maybe just `os.RemoveAll(downloadDir)`? The downloader will recreate it as necessary.